daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1090,6 +1090,29 @@ private[lf] object SBuiltin {
     }
   }
 
+  // SBPromoteAnyContract(templateId): ContractId actualTemplateId -> Any -> templateId
+  // Crashes unless actualTemplateId is a predecessor of templateId.
```

Comment:
probably a big flashing TODO 16151 should be added here

Reply:
I've `s/Crashes/Fails/`, since this is an expected failure. I did add a big TODO just below for covering non-structurally-equivalent types though, so thanks for that!